include/fiction/networks/virtual_pi_network.hpp

```diff
@@ -0,0 +1,505 @@
+        static_assert(fiction::has_foreach_real_pi_v<Ntk>, "Ntk does not implement the foreach_pi method");
+
+        mockturtle::node_map<mockturtle::signal<Ntk>, Ntk> old2new(src);
+        Ntk                                                dest;
+        old2new[src.get_constant(false)] = dest.get_constant(false);
+        if (src.get_node(src.get_constant(true)) != src.get_node(src.get_constant(false)))
+        {
+            old2new[src.get_constant(true)] = dest.get_constant(true);
+        }
+        src.foreach_real_pi([&](auto const& n) { old2new[n] = dest.create_pi(); });
+        return {dest, old2new};
+    }
+};
+}  // namespace detail
+
+/**
+ * Deletes virtual primary inputs from a network. This can mainly be used for equivalence checking.
+ * If the network does not have any virtual PIs stored, the network is returned.
+ *
+ * @tparam Ntk The type of network.
+ * @param ntk The input network.
+ * @return The resulting network after virtual primary inputs are deleted.
+ */
+template <typename Ntk>
+auto delete_virtual_pis(const Ntk& ntk) -> decltype(std::declval<Ntk>().clone())
+{
+    static_assert(mockturtle::is_network_type_v<Ntk>, "Ntk is not a network type");
+    static_assert(mockturtle::has_clone_v<Ntk>, "Ntk does not implement the clone constructor");
+    static_assert(mockturtle::has_get_node_v<Ntk>, "Ntk does not implement the get_node function");
+    static_assert(mockturtle::has_is_complemented_v<Ntk>, "Ntk does not implement the is_complemented function");
+    static_assert(mockturtle::has_foreach_pi_v<Ntk>, "Ntk does not implement the foreach_pi function");
+    static_assert(mockturtle::has_foreach_gate_v<Ntk>, "Ntk does not implement the foreach_gate function");
+    static_assert(mockturtle::has_foreach_po_v<Ntk>, "Ntk does not implement the foreach_po function");
+    static_assert(mockturtle::has_foreach_fanin_v<Ntk>, "Ntk does not implement the foreach_fanin function");
+    static_assert(mockturtle::has_get_constant_v<Ntk>, "Ntk does not implement the get_constant function");
+    static_assert(mockturtle::has_create_pi_v<Ntk>, "Ntk does not implement the create_pi function");
+    static_assert(mockturtle::has_create_po_v<Ntk>, "Ntk does not implement the create_po function");
+    static_assert(mockturtle::has_create_not_v<Ntk>, "Ntk does not implement the create_not function");
+    static_assert(fiction::has_get_real_pi_v<Ntk>, "Ntk does not implement the get_real_pi function");
+    static_assert(fiction::has_is_virtual_pi_v<Ntk>, "Ntk does not implement the is_vitual_pi function");
+    static_assert(fiction::has_num_real_pis_v<Ntk>, "Ntk does not implement the has_num_real_pis function");
+    static_assert(fiction::has_num_virtual_pis_v<Ntk>, "Ntk does not implement the num_virtual_pis function");
+    static_assert(fiction::has_foreach_real_pi_v<Ntk>, "Ntk does not implement the foreach_real_pi function");
```

Comment:
I have a feeling that this high number of ``static_asserts`` indicates that we are missing something in ``traits.hpp``.

Would it help to define something like ``fiction::virtual_pi_network_v<Ntk>``?